compiler/rustc_parse/src/parser/item.rs

```diff
@@ -2066,7 +2069,42 @@ impl<'a> Parser<'a> {
         if has_bang {
             self.expect(&token::Not)?; // `!`
         }
-        let ident = self.parse_ident()?;
+
+        let ident = match self.parse_ident() {
+            Ok(ident) => ident,
+            Err(mut err) => {
+                match (
+                    &self.token.kind,
+                    self.look_ahead(1, |token| token.ident()),
+                    self.look_ahead(2, |token| {
+                        (token.kind == TokenKind::CloseDelim(Delimiter::Parenthesis))
+                            .then_some(token.span)
+                    }),
+                ) {
+                    (
+                        TokenKind::OpenDelim(Delimiter::Parenthesis),
+                        Some((Ident { name, .. }, _)),
+                        Some(closing_span),
+                    ) => {
```

Comment:
Could you prefer a let chain here?

Reply:
Sure, I changed it to a let chain, it seems quite a bit cleaner. Thank you for the suggestion!